modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/ActionExecutorManagerImpl.java

```diff
@@ -97,7 +110,14 @@ private String _getActionExecutorKey(KaleoAction kaleoAction) {
 			return actionType.name();
 		}
 
-		return kaleoAction.getScriptLanguage();
+		String scriptLanguage = kaleoAction.getScriptLanguage();
+
+		if (Objects.equals(ScriptLanguage.JAVA.getValue(), scriptLanguage)) {
+			return scriptLanguage + StringPool.COLON +
+				StringUtil.trim(kaleoAction.getScript());
```

Comment:
@W2jx Is `kaleoAction.getScript()` always the class name of the action executor?

Reply:
Hi @dantewang , I used the original logic.